### PR TITLE
Add version to objects

### LINKED ISSUE #37: Downloading the data .tar does not handle deleted items
It overwrites, but does not delete, so you can still end up with ghost objects and transitions that have been deleted from the data repository.

### PR DESCRIPTION
**Pull request by ryanb**
* Objects are now sorted by most recent version instead of complexity
* `node process download` now clones `OneLifeData7` git repo instead of downloading `.tar`
  * Fixes #37
  * `git pull` is used to bring it up to date if it exists
* Git tags are used to populate object version numbers
  * `git tag -l` is used to list the tags
  * `git --name-status OneLife_v71..OneLife_v73` is used to see file changes
  * This is very fast so it can be done every process run
* Absolute paths are used in process script so it can run outside of process dir
  * `node process dev` works in main directory!

### CODE REVIEW

File: process/process.js [hunk 1]
```diff
@@ -1,17 +1,22 @@
 const GameData = require('./src/GameData');
 
-const gameData = new GameData();
+const gameData = new GameData(__dirname);
 
 if (process.argv.includes('download')) {
   console.log("Downloading data...");
-  gameData.download();
+  gameData.download("https://github.com/jasonrohrer/OneLifeData7.git");
```
Comment (Kazetsukai) [inline process/process.js]: If we can pull this out to a const up the top, will be a little nicer for UncleGus to keep his fork in sync.

Comment (ryanb) [reply process/process.js]: How about using an ENV variable which falls back to this? Then no code change will be necessary. I'll try this.

File: process/src/VersionPopulator.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  tagFromVersion(version) {
+    if (version == 0) return "OneLife_vStart";
+    return "OneLife_v" + version;
+  }
+
+  populateVersion(current, previous) {
+    const differences = this.diff(current, previous);
+    for (let difference of differences) {
+      if (difference[0] == "A")
+        this.populateVersionAddition(current, difference[1]);
```
[the remark this answers is not in the document]
Comment (ryanb) [reply process/src/VersionPopulator.js]: Right, it's easy to determine if an object has been updated. However determining what was updated means we need to parse the old file. That could get hairy if the object format ever changes and we need to support legacy formats.

[the remark this answers is not in the document]
Comment (Kazetsukai) [reply process/src/VersionPopulator.js]: Definitely. Suddenly everything is "updated yesterday".

File: process/process.js [hunk 3]
```diff
@@ -1,10 +1,12 @@
 const GameData = require('./src/GameData');
 
+const gitURL = process.env.ONETECH_PROCESS_GIT_URL || "https://github.com/jasonrohrer/OneLifeData7.git";
```
Comment (Kazetsukai) [inline process/process.js]: @UncleGus this should mean you just need to pull latest to your fork of Onetech, and set the environment variable to your OneLifeData repo when processing

**Pull request merged**